### PR TITLE
feat: add service_credential secret type

### PR DESCRIPTION
**Pull request by alex-reiff**
### Description

Add support to create a service_credential type secret in secrets manager.

### Release required?

- [ ] No release
- [ ] Patch release (`x.x.X`)
- [x] Minor release (`x.X.x`)
- [ ] Major release (`X.x.x`)

##### Release notes content

Support has been added for creating `Service credentials` secrets in Secrets Manager. 

Before you start working with them, you will need to create a service authorization between Secrets Manager and the target service, see the provided example for more info.

Variables and outputs related to auto_rotation have been renamed. This may impact existing users of "user_password" type secrets.

Variables:
secret_user_pass_auto_rotation -> secret_auto_rotation
secret_user_pass_auto_rotation_unit -> secret_auto_rotation_unit
secret_user_pass_auto_rotation_interval -> secret_auto_rotation_interval

Outputs:
user_pass_rotation -> secret_auto_rotation
user_pass_auto_rotation_interval -> secret_rotation_interval
user_pass_next_rotation_date -> secret_next_rotation_date

### Run the pipeline

If the CI pipeline doesn't run when you create the PR, the PR requires a user with GitHub collaborators access to run the pipeline.

Run the CI pipeline when the PR is ready for review and you expect tests to pass. Add a comment to the PR with the following text:

```
/run pipeline
```

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### For mergers

- Use a conventional commit message to set the release level. Follow the [guidelines](https://terraform-ibm-modules.github.io/documentation/#/merging.md).
- Include information that users need to know about the PR in the commit message. The commit message becomes part of the GitHub release notes.
- Use the **Squash and merge** option.

### CODE REVIEW

**Comment by alex-reiff** [comment]
Skipping upgrade test. I have changed the default secret_auto_rotation value to 89 (because service credentials have a max TTL of 90 and auto_rotate has to be strictly shorter). Because I also renamed this variable, I can't specify it in the unit test to make the upgrade test pass.

Here is total diff that it causing the failure:
```
TestRunUpgradeExample 2024-01-18T18:55:20Z command.go:185:   ~ resource "ibm_sm_username_password_secret" "username_password_secret" {
TestRunUpgradeExample 2024-01-18T18:55:20Z command.go:185:         id                 = "us-south/79c6d411-c18f-4670-b009-b0044a238667/11358800-ec64-8ece-add3-6756b128d0ee"
TestRunUpgradeExample 2024-01-18T18:55:20Z command.go:185:         name               = "sm-secret-module-upg-i9whcg-user-pass-secret"
TestRunUpgradeExample 2024-01-18T18:55:20Z command.go:185:         # (21 unchanged attributes hidden)
TestRunUpgradeExample 2024-01-18T18:55:20Z command.go:185: 
TestRunUpgradeExample 2024-01-18T18:55:20Z command.go:185:       ~ rotation {
TestRunUpgradeExample 2024-01-18T18:55:20Z command.go:185:           ~ interval    = 90 -> 89
TestRunUpgradeExample 2024-01-18T18:55:20Z command.go:185:             # (2 unchanged attributes hidden)
TestRunUpgradeExample 2024-01-18T18:55:20Z command.go:185:         }
TestRunUpgradeExample 2024-01-18T18:55:20Z command.go:185:     }
```

File: examples/complete/main.tf [hunk 1]
```diff
@@ -190,3 +190,42 @@ module "secret_manager_imported_cert" {
...
+  kms_encryption_enabled = false
+}
+
+#create a service authorization between Secrets Manager and the target service (COS)
+resource "ibm_iam_authorization_policy" "policy" {
+  source_service_name         = "secrets-manager"
+  target_service_name         = "cloud-object-storage"
+  target_resource_instance_id = module.cloud_object_storage.cos_instance_guid
+  roles                       = ["Key Manager"]
+}
```
Comment (ocofaigh) [inline examples/complete/main.tf:217]: question for maybe the playback next week - should the module support creating the auth policy if its going to support creating the service credentials

**Review by shemau — CHANGES_REQUESTED** [review]
The code looks good, a few comments on the readme.

Additionally, the description of the PR, under release notes content should include the inputs being renamed. This may impact existing users of "user_password" type secrets.

secret_user_pass_auto_rotation becomes secret_auto_rotation
secret_user_pass_auto_rotation_unit becomes secret_auto_rotation_unit
secret_user_pass_auto_rotation_interval becomes secret_auto_rotation_interval

The outputs are not renamed (maybe they should be).

File: examples/complete/main.tf (hunk 1 above)
Comment (ocofaigh) [reply examples/complete/main.tf:219]: @alex-reiff do you remember what we decided to do here?

Comment (alex-reiff) [reply examples/complete/main.tf:219]: I believe we said that this is fine for the example. We don't have any functionality to add it in the module itself.

**Review by shemau — APPROVED** [review]
LGTM

**Pull request merged**